src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs

```diff
@@ -64,6 +65,10 @@ public RecordType(DisplayNameProvider displayNameProvider, TableDelegationInfo d
 
         public override IEnumerable<string> FieldNames => _fieldNames;
 
+        // List of fields names that compose the primary key
+        // This array is ordered and contains logical names
+        public virtual string[] PrimaryKeyNames => Array.Empty<string>();
```

Comment:
>PrimaryKeyNames [](http://example.com/codeflow?start=32&length=15)

It looks like we have something similar already, can we reuse this: 

[https://github.com/microsoft/Power-Fx/blob/46a1bfc11a173fd22c04f3b0b2ae7a55b72e6368/src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs#L203](https://github.com/microsoft/Power-Fx/blob/46a1bfc11a173fd22c04f3b0b2ae7a55b72e6368/src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs#L203)

Reply:
I'll move that one to AggregateType
This one is different in the sense that we need to support composed keys